python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -2036,6 +2106,11 @@ def fault(self, code=None, report=""):
         disable_corr.disableAll = True
         self.mark_corrections(disable_corr, False)
 
+        # Now lower the mirrors if the corrections were enabled
+        await self.lower_mirrors_to_hardpoints(
```

Comment:
I see why you wanted to make fault a coroutine. I don't think salobj supports it though and fault must be a regular method. What you can do in this case is schedule the coroutine to run in the background, e.g.;

```
self.lower_mirrors_task = asyncio.create_task(self.lower_mirrors_to_hardpoints(m1=self.corrections["m1"], m2=self.corrections["m2"]))
```

In this case you probably want to add 

```
self.lower_mirrors_task = salobj.make_done_future()
```

to `__init__`. Then, you could probably add something like this in `handle_summary_state`:

```
if not self.lower_mirrors_task.done():
    raise RuntimeError("Mirror still being lowered.")
else:
    try:
        await self.lower_mirrors_task
    except Exception:
         self.log.exception("Exception while lowering mirrors. Ignoring.")
```

Then in this method you probably want to do something like:

```
if self.lower_mirrors_task.done():
    self.lower_mirrors_task = asyncio.create_task(self.lower_mirrors_to_hardpoints(m1=self.corrections["m1"], m2=self.corrections["m2"]))
else:
    self.log.info("Mirrors already being lowered. Skipping...")
```

Reply:
so i get your argument for fault state, that's been implemented (I think... it doesn't actually get tested)
I don't quite understand your suggestion for `handle_summary_state` and why it's necessary. First, this csc doesn't use a `handle_summary_state` method and instead uses `end_disable` - but even inside `end_disable` why does the task have to run in the background can't it just be run as is (line 543)